tracer/test/test-applications/integrations/Samples.SqlServer/StoredProcedures.cs

```diff
@@ -0,0 +1,213 @@
+        {
+            // Similar to DbCommandFactory.GetSelectRowCommand
+            Console.WriteLine("Test dbo.sp_GetTableRow");
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandText = $@"
+CREATE PROCEDURE dbo.sp_GetTableRow
+    @Id int
+AS
+BEGIN
+    SELECT * FROM {tableName} WHERE Id = @Id;
+END";
+                await command.ExecuteNonQueryAsync(token);
+            }
+
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandTimeout = 5; // running into issues where something is hanging
+                command.CommandType = CommandType.StoredProcedure;
+                command.CommandText = "dbo.sp_GetTableRow";
+
+                var param = command.CreateParameter();
+                param.ParameterName = "@Id";
+                param.Value = 2;
+                command.Parameters.Add(param);
+
+                using var reader = await command.ExecuteReaderAsync(token);
+                while (await reader.ReadAsync(token))
+                {
+                    // consume the reader otherwise it locks
+                }
+            }
+        }
+
+        private static async Task TestOutputParameters(SqlConnection connection, string tableName, CancellationToken token)
+        {
+            Console.WriteLine("Test dbo.sp_UpdateRowWithOutput");
+            // Similar to DbCommandFactory.GetUpdateRowCommand but with OUTPUT parameters
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandText = $@"
+CREATE PROCEDURE dbo.sp_UpdateRowWithOutput
+    @Id int,
+    @NewName varchar(100),
+    @OldName varchar(100) OUTPUT
+AS
+BEGIN
+    -- Store the old name before updating
+    SELECT @OldName = Name FROM {tableName} WHERE Id = @Id;
+    
+    -- Update the row
+    UPDATE {tableName} SET Name = @NewName WHERE Id = @Id;
+    
+    -- Return number of rows affected
+    RETURN @@ROWCOUNT;
+END";
+                await command.ExecuteNonQueryAsync(token);
+            }
+
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandTimeout = 5; // running into issues where something is hanging
+
+                command.CommandType = CommandType.StoredProcedure;
+                command.CommandText = "dbo.sp_UpdateRowWithOutput";
+
+                // Input parameters - should be added to EXEC command
+                command.Parameters.AddWithValue("@Id", 1);
+                command.Parameters.AddWithValue("@NewName", "UpdatedName1");
+
+                // Output parameter - shoudl be added to EXEC command
+                var oldNameParam = command.Parameters.AddWithValue("@OldName", DBNull.Value);
+                oldNameParam.Direction = ParameterDirection.Output;
+                oldNameParam.Size = 100;
```

Comment:
Is there any validation we can do here to make sure that the OUTPUT parameter and the return value are unaffected?

Reply:
What I'm wondering in general is how we validate that the stored procedure ran as expected. I see that we now have spans for those operations, but since we're doing some rewriting, can we either validate the output from the stored procedure call or should we rely on our unit tests to assert what the rewritten query looks like?

Separately, we'll need to validate that the DBM correlation looks correct in the UI, but that should be able to be done quickly and separately from the PR review.

Reply:
> Separately, we'll need to validate that the DBM correlation looks correct in the UI, but that should be able to be done quickly and separately from the PR review.

Yes I need to figure out how to do that still